frontend_build/src/intl_code_gen.js

```diff
@@ -0,0 +1,68 @@
+const supportedLanguages = require('../../kolibri/locale/supported_languages.json');
+
+const vueIntlHeader = `module.exports = function (locale) {
+  switch (locale) {`;
+
+const generateVueIntlItems = language => {
+  return `    case '${language.language_code}${
+    language.language_territory ? '-' + language.language_territory.toUpperCase() : ''
+  }':
+      return new Promise(function (resolve) {
+        require.ensure(
+          ['vue-intl/locale-data/${language.language_code}.js'],
+          function (require) {
+            resolve(require('vue-intl/locale-data/${language.language_code}.js'));
+          }
+        );
+      });`;
+};
+
+const vueIntlFooter = `    default:
+      return Promise.resolve({});
+  }
+};
+`;
+
+const vueIntlModule =
+  vueIntlHeader + supportedLanguages.map(generateVueIntlItems).join('') + vueIntlFooter;
+
+const intlHeader = `module.exports = function(locale) {
+  switch (locale) {`;
+
+const generateIntlItems = language => {
+  return `    case '${language.language_code}${
+    language.language_territory ? '-' + language.language_territory.toUpperCase() : ''
+  }':
+      return new Promise(function(resolve) {
+        require.ensure(
+          ['intl/locale-data/jsonp/${language.language_code}${
+    language.language_territory ? '-' + language.language_territory.toUpperCase() : ''
+  }.js'],
+          function(require) {
+            resolve(require('intl/locale-data/jsonp/${language.language_code}${
+    language.language_territory ? '-' + language.language_territory.toUpperCase() : ''
+  }.js'));
+          }
+        );
+      });`;
+};
+
+const intlFooter = `    default:
+      return new Promise(function(resolve) {
+        require.ensure(
+          ['intl/locale-data/jsonp/en.js'],
+          function(require) {
+            resolve(require('intl/locale-data/jsonp/en.js'));
+          }
+        );
+      });
+  }
+};
+`;
+
+const intlModule = intlHeader + supportedLanguages.map(generateIntlItems).join('') + intlFooter;
+
+module.exports = {
+  intlModule,
+  vueIntlModule,
+};
```

Comment:
File-generation by inlining javascript code strings is a bit difficult to read and understand.

It might be cleaner and less brittle to use a simple templating language like lodash templates or handlebars and store the template as an external file?

If that seems like too much effort, I'd recommend at least pasting a representative example from the old _import-intl-locale.js_ into a comment at the top so it's more clear what's being generated.

Reply:
I did initially use lodash templates for this, but it felt more unreadable to me. Can define some intermediate variables to improve readability and add some examples.